docs/DeveloperGuide.md

```diff
@@ -283,33 +283,97 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 ### Use cases
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
-
-**Use case: Delete a person**
-
-**MSS**
-
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+(For all use cases below, the **System** is the `UNion Application` and the **Actor** is the `user`, unless specified otherwise)
+
+**System: UNion Application (UNa)**\
+**Use case: UC01 - Add a new folder**\
+**Actor: User**\
+**Guarantees:**
+   * A new folder with the specified name is created as part of the application.
+
+**MSS:**
+   1. User keys in folder name after `mkdir` keyword.
+
+   Use case ends.
+
+**Extensions:**
+   > 1a. UNa detects error in the format specified by user.
+      >> 1a1. No new folder is created\
+      >> 1a2. User keys folder name `mkdir` keyword.\
+      >> Steps 1a1-1a2 are repeated until the data entered are correct.
+
+**System: UNion Application (UNa)**\
+**Use case: UC02 - Add a new contact to UNa**\
+**Actor: User**\
+**Guarantees:**
+   * A new contact with the specified details is stored as a standalone in the application.
+
+**MSS:**
+
+   1. User inputs contact details after `Add` keyword.
+
+   Use case ends.
+
+**Extensions:**
+   > 1a. UNa detects error in format given by user.
+      >> 1a1. No new contact was registered by UNa\
+      >> 1a2. User inputs details after `Add` keyword.\
+      >> Steps 1a1-1a2 are repeated until the data format entered are correct.
+
+**System: UNion Application (UNa)**\
+**Use case: UC03 - Add existing contact to a existing folder**\
+**Actor: User**\
+**Preconditions:** Folder and contact already exists in UNa\
+**MSS:**
+
+   1. <u> User adds a new folder named A (UC01) </u>
+   2. <u> User adds a new contact named Clarence (UC02) </u>
+   3. User specifies the contact index to be added to the named folder.
+
+   Use case ends.
+
+**Extensions:**
+   > 3a. User inputs an index greater than number of contacts available.
```

Comment:
perhaps it would be more accurate to say "invalid index"? since this extension can also include negative numbers or even non-integers